---
title: "Invertebrate methylomes provide insight into mechanisms of environmental tolerance and reveal methodological biases"
type: "article"
author: "Shelly A. Wanamaker , Yaamini R. Venkataraman, Mackenzie R. Gavery, Steven B. Roberts, Debashish Bhattacharya, Alan Downey-Wall, Jose M. Eirin-Lopez, Kevin M. Johnson, Katie E. Lotterhos, Jonathan B. Puritz and Hollie M. Putnam"
year: "2021"
publication: "Molecular Ecology Resources"
preprint: "https://www.biorxiv.org/content/10.1101/2021.03.29.437539v2"
doi: "10.1111/1755-0998.13542"
materials: "http://doi.org/10.17605/OSF.IO/X5WAZ"
eprint: "https://d.pr/f/LqAVdC"
toc: FALSE
categories:
  - Open Data Badge
  - coral
  - methylation
  - epigenetic
---

## Citation

> Wanamaker , S. A., Venkataraman, Y. R., Gavery, M. R., Roberts, S. B., Bhattacharya, D., Downey-Wall, A., Eirin-Lopez, J. M., Johnson, K. M., Lotterhos, K. E., Puritz, J. B., & Putnam, H. M. (2022). Invertebrate methylomes provide insight into mechanisms of environmental tolerance and reveal methodological biases. Molecular Ecology Resources, 22, 1247-- 1261. https://doi.org/10.1111/1755-0998.13542

## Abstract

There is a growing focus on the role of DNA methylation in the ability of marine invertebrates to rapidly respond to changing environmental factors and anthropogenic impacts. However, genome-wide DNA methylation studies in nonmodel organisms are currently hampered by a limited understanding of methodological biases. Here, we compare three methods for quantifying DNA methylation at single base-pair resolution---whole genome bisulfite sequencing (WGBS), reduced representation bisulfite sequencing (RRBS), and methyl-CpG binding domain bisulfite sequencing (MBDBS)---using multiple individuals from two reef-building coral species with contrasting environmental sensitivity. All methods reveal substantially greater methylation in Montipora capitata (11.4%) than the more sensitive Pocillopora acuta (2.9%). The majority of CpG methylation in both species occurs in gene bodies and flanking regions. In both species, MBDBS has the greatest capacity for detecting CpGs in coding regions at our sequencing depth, but MBDBS may be influenced by intrasample methylation heterogeneity. RRBS yields robust information for specific loci albeit without enrichment of any particular genome feature and with significantly reduced genome coverage. Relative genome size strongly influences the number and location of CpGs detected by each method when sequencing depth is limited, illuminating nuances in cross-species comparisons. As genome-wide methylation differences, supported by data across bisulfite sequencing methods, may contribute to environmental sensitivity phenotypes in critical marine invertebrate taxa, these data provide a genomic resource for investigating the functional role of DNA methylation in environmental tolerance.

## Data Availability

::: callout-important
## Open Data

This article has earned an Open Data Badge for making publicly available the digitally-shareable data necessary to reproduce the reported results
:::

The data sets supporting the conclusions of this article are available in the Coral Methylation Methods Comparison repository, http://doi.org/10.17605/OSF.IO/X5WAZ, and included within the article and its additional files. All raw data can be accessed under NCBI Bioproject PRJNA691891.
