<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>robertslab.info - Research</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">robertslab.info</span>
    </a>
  </div>
        <div class="quarto-navbar-tools ms-auto tools-wide">
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label=""><i class="bi bi-hdd-network"></i></a>
      <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item quarto-navbar-tools ms-auto-item" href="https://github.com/RobertsLab/resources/issues/new">
            raven
            </a>
          </li>
          <li>
            <a class="dropdown-item quarto-navbar-tools ms-auto-item" href="https://github.com/RobertsLab">
            gannet
            </a>
          </li>
      </ul>
    </div>
    <a href="" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-journal-code"></i></a>
    <a href="" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-calendar-week"></i></a>
    <a href="https://calendar.google.com/calendar/embed?src=mrc305@gmail.com&amp;ctz=America/Vancouver" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-"></i></a>
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-1" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label=""><i class="bi bi-github"></i></a>
      <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="quarto-navigation-tool-dropdown-1">
          <li>
            <a class="dropdown-item quarto-navbar-tools ms-auto-item" href="https://github.com/RobertsLab/resources/issues/new">
            Submit an Issue
            </a>
          </li>
          <li>
            <a class="dropdown-item quarto-navbar-tools ms-auto-item" href="https://github.com/RobertsLab">
            Lab Organization
            </a>
          </li>
      </ul>
    </div>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./research.html">Research</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Research</h1>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./research.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Research</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Peaple</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./people.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab Members</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./collaborators.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Collaborators</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Products</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./publications02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Papers</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./publications.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Publications</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./products.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Presentations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">More</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./posts.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">News and Updates</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lab Culture</span>
  </li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#collaborative-research-urol-epigenetics-2-predicting-phenotypic-and-eco-evolutionary-consequences-of-environmental-energetic-epigenetic-linkages" id="toc-collaborative-research-urol-epigenetics-2-predicting-phenotypic-and-eco-evolutionary-consequences-of-environmental-energetic-epigenetic-linkages" class="nav-link active" data-scroll-target="#collaborative-research-urol-epigenetics-2-predicting-phenotypic-and-eco-evolutionary-consequences-of-environmental-energetic-epigenetic-linkages">Collaborative Research: URoL : Epigenetics 2: Predicting phenotypic and eco-evolutionary consequences of environmental-energetic-epigenetic linkages</a></li>
  <li><a href="#collaborative-research-does-ocean-acidification-induce-a-methylation-response-that-affects-the-fitness-of-the-next-generation-in-oysters" id="toc-collaborative-research-does-ocean-acidification-induce-a-methylation-response-that-affects-the-fitness-of-the-next-generation-in-oysters" class="nav-link" data-scroll-target="#collaborative-research-does-ocean-acidification-induce-a-methylation-response-that-affects-the-fitness-of-the-next-generation-in-oysters">Collaborative Research: Does ocean acidification induce a methylation response that affects the fitness of the next generation in oysters?</a></li>
  <li><a href="#elucidating-the-physiological-and-epigenetic-response-of-tetraploid-and-triploid-pacific-oysters-crassostrea-gigas-to-environmental-stressors" id="toc-elucidating-the-physiological-and-epigenetic-response-of-tetraploid-and-triploid-pacific-oysters-crassostrea-gigas-to-environmental-stressors" class="nav-link" data-scroll-target="#elucidating-the-physiological-and-epigenetic-response-of-tetraploid-and-triploid-pacific-oysters-crassostrea-gigas-to-environmental-stressors">Elucidating the physiological and epigenetic response of tetraploid and triploid Pacific Oysters (Crassostrea gigas) to environmental stressors</a></li>
  <li><a href="#improving-genomic-resources-to-support-restoration-and-protection-of-the-olympia-oyster-in-puget-sound" id="toc-improving-genomic-resources-to-support-restoration-and-protection-of-the-olympia-oyster-in-puget-sound" class="nav-link" data-scroll-target="#improving-genomic-resources-to-support-restoration-and-protection-of-the-olympia-oyster-in-puget-sound">Improving genomic resources to support restoration and protection of the Olympia Oyster in Puget Sound</a></li>
  <li><a href="#applying-cutting-edge-technology-for-reproductive-control-in-emerging-bivalve-species" id="toc-applying-cutting-edge-technology-for-reproductive-control-in-emerging-bivalve-species" class="nav-link" data-scroll-target="#applying-cutting-edge-technology-for-reproductive-control-in-emerging-bivalve-species">Applying cutting-edge technology for reproductive control in emerging bivalve species</a></li>
  <li><a href="#enhancing-sustainability-of-shellfish-aquaculture-through-streamlined-maturation-control" id="toc-enhancing-sustainability-of-shellfish-aquaculture-through-streamlined-maturation-control" class="nav-link" data-scroll-target="#enhancing-sustainability-of-shellfish-aquaculture-through-streamlined-maturation-control">Enhancing sustainability of shellfish aquaculture through streamlined maturation control</a></li>
  <li><a href="#effects-of-temperature-change-and-hematodinium-sp.-infection-bitter-crab-disease-on-tanner-crab-chionoecetes-bairdi" id="toc-effects-of-temperature-change-and-hematodinium-sp.-infection-bitter-crab-disease-on-tanner-crab-chionoecetes-bairdi" class="nav-link" data-scroll-target="#effects-of-temperature-change-and-hematodinium-sp.-infection-bitter-crab-disease-on-tanner-crab-chionoecetes-bairdi">Effects of temperature change and Hematodinium sp. infection (Bitter Crab Disease) on Tanner crab (Chionoecetes bairdi)</a></li>
  <li><a href="#development-of-environmental-conditioning-practices-to-decrease-impacts-of-climate-change-on-shellfish-aquaculture" id="toc-development-of-environmental-conditioning-practices-to-decrease-impacts-of-climate-change-on-shellfish-aquaculture" class="nav-link" data-scroll-target="#development-of-environmental-conditioning-practices-to-decrease-impacts-of-climate-change-on-shellfish-aquaculture">Development of environmental conditioning practices to decrease impacts of climate change on shellfish aquaculture</a></li>
  <li><a href="#caliguslife-scientific-research-of-excellence-towards-understanding-sea-lice-biology-and-its-application-in-control-strategies-for-the-salmon-industry" id="toc-caliguslife-scientific-research-of-excellence-towards-understanding-sea-lice-biology-and-its-application-in-control-strategies-for-the-salmon-industry" class="nav-link" data-scroll-target="#caliguslife-scientific-research-of-excellence-towards-understanding-sea-lice-biology-and-its-application-in-control-strategies-for-the-salmon-industry">CaligusLIFE: Scientific research of excellence towards understanding sea lice biology and its application in control strategies for the salmon industry</a></li>
  <li><a href="#leveraging-transformative-omics-technologies-to-alleviate-barriers-to-us-shellfish-production-omics-tech" id="toc-leveraging-transformative-omics-technologies-to-alleviate-barriers-to-us-shellfish-production-omics-tech" class="nav-link" data-scroll-target="#leveraging-transformative-omics-technologies-to-alleviate-barriers-to-us-shellfish-production-omics-tech">Leveraging transformative ’omics technologies to alleviate barriers to US shellfish production (Omics Tech)</a></li>
  <li><a href="#development-of-omics-and-bioinformatics-approaches-for-marine-organisms-in-support-of-research-in-aquaculture-ocean-acidification-and-fisheries-assessments-aqua-omics" id="toc-development-of-omics-and-bioinformatics-approaches-for-marine-organisms-in-support-of-research-in-aquaculture-ocean-acidification-and-fisheries-assessments-aqua-omics" class="nav-link" data-scroll-target="#development-of-omics-and-bioinformatics-approaches-for-marine-organisms-in-support-of-research-in-aquaculture-ocean-acidification-and-fisheries-assessments-aqua-omics">Development of ’omics and bioinformatics approaches for marine organisms in support of research in aquaculture, ocean acidification, and fisheries assessments (Aqua-Omics)</a></li>
  <li><a href="#development-of-innovative-approaches-to-support-sustainable-aquaculture-and-understand-the-effects-of-ocean-acidification-on-marine-species-vernon" id="toc-development-of-innovative-approaches-to-support-sustainable-aquaculture-and-understand-the-effects-of-ocean-acidification-on-marine-species-vernon" class="nav-link" data-scroll-target="#development-of-innovative-approaches-to-support-sustainable-aquaculture-and-understand-the-effects-of-ocean-acidification-on-marine-species-vernon">Development of innovative approaches to support sustainable aquaculture and understand the effects of ocean acidification on marine species (Vernon)</a></li>
  <li><a href="#readying-sustainable-aquaculture-for-a-changing-ocean-uncovering-the-mechanisms-associated-with-intergenerational-carryover-effects-to-enhance-bivalve-resilience-to-acidification" id="toc-readying-sustainable-aquaculture-for-a-changing-ocean-uncovering-the-mechanisms-associated-with-intergenerational-carryover-effects-to-enhance-bivalve-resilience-to-acidification" class="nav-link" data-scroll-target="#readying-sustainable-aquaculture-for-a-changing-ocean-uncovering-the-mechanisms-associated-with-intergenerational-carryover-effects-to-enhance-bivalve-resilience-to-acidification">Readying sustainable aquaculture for a changing ocean: uncovering the mechanisms associated with intergenerational carryover effects to enhance bivalve resilience to acidification</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>Research in our lab focuses on characterizing physiological responses of marine organisms to environmental change. Using integrative approaches we strive to examine impacts and adaptive potential from the nucleotide to organismal level. A core component of this includes investigating the functional relationship of genetics, epigenetics, and protein expression. Three major themes of are current projects include <strong>Environmental Epigenetics</strong>, <strong>Reproductive Biology and Ecology</strong>, and <strong>Aquaculture Resource Development</strong>.</p>
<blockquote class="blockquote">
<h1 id="environmental-epigenetics">Environmental Epigenetics</h1>
</blockquote>
<section id="collaborative-research-urol-epigenetics-2-predicting-phenotypic-and-eco-evolutionary-consequences-of-environmental-energetic-epigenetic-linkages" class="level3">
<h3 class="anchored" data-anchor-id="collaborative-research-urol-epigenetics-2-predicting-phenotypic-and-eco-evolutionary-consequences-of-environmental-energetic-epigenetic-linkages">Collaborative Research: URoL : Epigenetics 2: Predicting phenotypic and eco-evolutionary consequences of environmental-energetic-epigenetic linkages</h3>
<p><em>Summary</em>: Living organisms may acclimate to environmental changes through epigenetic modifications to DNA, which alter the way genetic instructions are interpreted without altering the DNA code itself. While these modifications to organismal phenotype or function can be reversible, some of them may be inherited by offspring, potentially producing multiple, heritable outcomes from a single genome and affecting ecological and evolutionary outcomes. This project uses symbiotic, metabolically complex reef building corals as a model system to test the connections between physiological, epigenetic, and metabolic states, and predict how population and community dynamics are influenced by epigenetically-modulated phenotypes. This work will advance biological knowledge by delineating fundamental links (Rules of Life) between ubiquitous organismal energetic processes, epigenetics, and eco-evolutionary outcomes. The Broader Impacts activities parallel the project’s integrative approach, linking insights from Environment x Energetics x Epigenetics x Ecology for Education into an E5 platform. The E5 platform will provide i) early career STEM training, ii) local and global community education, and iii) educational resources for open science, quantitative approaches, and research reproducibility. Further, this E5 platform will train and inform the next generation of diverse scientists and public by combining local and global initiatives focusing on groups underrepresented in STEM.</p>
<p>This project examines how nutrient metabolism in the mitochondria generates cofactors and energy that will instruct the epigenetic machinery in the cell nucleus to modulate genome function to appropriately respond to environmental conditions. Environmentally-responsive metabolic function and energetic-epigenetic linkages act as drivers of complex emergent phenotypes. To elucidate relationships that are the basis for Rules of Life with respect to epigenetics, this project will use integrative experimental and modeling approaches focused on reef building corals to: 1) link nutritionally-provisioned metabolites with epigenetic and organismal state through seasonal sampling across environmental gradients; 2) expand current Dynamic Energy Budget (DEB) models for symbiotic organisms to further integrate critical facets of nutritional symbiosis and calcification; 3) experimentally modulate metabolic and therefore epigenetic states through repeated exposure to increased temperature and nutrients, to test intra- and trans-generational epigenetic inheritance; 4) use DEB theory to identify shifts in energetics associated with epigenetic modulation, and link these sub-organismal processes to higher levels of organization; and 5) integrate findings into a generalizable, predictive eco-evolutionary model that links nutritional interactions, metabolic states, and subsequent epigenetic effects to the timescales regulating organismal processes and eco-evolutionary outcomes. This effort will provide characterization of environmental epigenetic phenomena in ecosystem-engineering marine invertebrates. This characterization includes determining the mechanisms and the degree of epigenetic ‘memory’ both within and across generations. By including information on environmental legacies, propagated by epigenetics, this project will advance both organismal and population-based models and improve capacity to predict responses to acute and chronic environmental signals.</p>
</section>
<section id="collaborative-research-does-ocean-acidification-induce-a-methylation-response-that-affects-the-fitness-of-the-next-generation-in-oysters" class="level3">
<h3 class="anchored" data-anchor-id="collaborative-research-does-ocean-acidification-induce-a-methylation-response-that-affects-the-fitness-of-the-next-generation-in-oysters">Collaborative Research: Does ocean acidification induce a methylation response that affects the fitness of the next generation in oysters?</h3>
<p><em>Summary</em>: Marine ecosystems worldwide are threatened by ocean acidification, a process caused by the unprecedented rate at which carbon dioxide is increasing in the atmosphere. Since ocean change is predicted to be rapid, extreme, and widespread, marine species may face an “adapt-or-die” scenario. However, modifications to the DNA sequence may be induced in response to a stress like ocean acidification and then inherited. Such “epigenetic” modifications may hold the key to population viability under global climate change, but they have been understudied. The aim of this research is to characterize the role of DNA methylation, a heritable epigenetic system, in the response of Eastern oysters (Crassostrea virginica) to ocean acidification. The intellectual merit lies in the integrative approach, which will characterize the role of DNA methylation in the intergenerational response of oysters to ocean acidification. These interdisciplinary data, spanning from molecular to organismal levels, will provide insight into mechanisms that underlie the capacity of marine invertebrates to respond to ocean acidification and lay the foundation for future transgenerational studies. Ocean acidification currently threatens marine species worldwide and has already caused significant losses in aquaculture, especially in Crassostrea species. This research has broader impacts for breeding, aquaculture, and the economy. Under the investigators’ “Epigenetics to Ocean” (E2O) training program, the investigators will build STEM talent in bioinformatics and biogeochemistry, expose girls in low-income school districts to careers in genomics, and advance the field through open science and reproducibility.</p>
</section>
<section id="elucidating-the-physiological-and-epigenetic-response-of-tetraploid-and-triploid-pacific-oysters-crassostrea-gigas-to-environmental-stressors" class="level3">
<h3 class="anchored" data-anchor-id="elucidating-the-physiological-and-epigenetic-response-of-tetraploid-and-triploid-pacific-oysters-crassostrea-gigas-to-environmental-stressors">Elucidating the physiological and epigenetic response of tetraploid and triploid Pacific Oysters (Crassostrea gigas) to environmental stressors</h3>
<p><em>Summary</em>: Basic physiological studies will be conducted with diploid, triploid and tetraploid larvae produced from the same MBP lines to determine tolerances to three environmental factors which are being altered by climate change. These include temperature, salinity and low pH, individually and in combination. Differential response will be assessed by changes in oxygen consumption rates, shell morphology and survival rates. Cohorts of diploid, triploid and tetraploid juvenile oysters will be grown out in several locations in the two States and closely monitored for survival and performance (e.g.&nbsp;growth rate, condition index). Shotgun proteomics will be used to permit efficient comparison of protein expression patterns on the entire genome. Changes in global DNA methylation will also be examined to assess the role of the environment in influencing the epigenetic landscape, and how this translates to phenotype. A subset of oysters from the first experiments will be selected for spawning larvae that can then be similarly assessed to determine how parental environmental conditions affect larval performance.</p>
</section>
<section id="improving-genomic-resources-to-support-restoration-and-protection-of-the-olympia-oyster-in-puget-sound" class="level3">
<h3 class="anchored" data-anchor-id="improving-genomic-resources-to-support-restoration-and-protection-of-the-olympia-oyster-in-puget-sound">Improving genomic resources to support restoration and protection of the Olympia Oyster in Puget Sound</h3>
<p><em>Summary</em>: There is a significant gap in our fundamental understanding of this species’ resilience in the face of environmental change, ecological interactions, and population structure. This information is critical to local restoration efforts and to predicting how molluscs will adapt to long-term environmental change. There is recent evidence that oysters have the capacity to respond to environmental change at a rate beyond what would be predicted by genetic variation alone. The overall objective of this research is to produce genomic resources and capacity to understand the response of Olympia oysters to environmental change. Specifically, a draft genome assembly for the Olympia oyster will be produced and used to understand how responses of the Olympia oyster to environmental changes are inherited (i.e., genetic or epigenetic) using restriction site associated DNA sequencing (RAD-Seq) and bisulfite sequencing (BS-Seq). A web-based platform will be developed based on these resources that will be used for discovery and further collaboration.</p>
<blockquote class="blockquote">
<h1 id="reproductive-biology-and-ecology">Reproductive Biology and Ecology</h1>
</blockquote>
</section>
<section id="applying-cutting-edge-technology-for-reproductive-control-in-emerging-bivalve-species" class="level3">
<h3 class="anchored" data-anchor-id="applying-cutting-edge-technology-for-reproductive-control-in-emerging-bivalve-species">Applying cutting-edge technology for reproductive control in emerging bivalve species</h3>
<p><em>Summary</em>: Given the supreme importance of maintaining ecological security in expanding shellfish farming to meet the global demand for environmentally sustainable protein, our goal is to engage a novel mix of disciplines to create new knowledge leading to an optimal approach for conferring sterility in emerging shellfish species. We have assembled an interdisciplinary team including aquaculture specialists, molecular physiologists, and single cell genomicists to tackle the critical initial step in this approach: the identification of genes involved in germ cell fate at the earliest developmental stages in geoducks. We anticipate identifying several germ cell- specific genes that when disrupted effectively turn off germ cell differentiation, yielding sterile products that do not undergo gonadogenesis, exhibit superior performance, and cannot disrupt genetic structure of ecosystems. Identification of these genes will be the measure of success for the proposed work. These gene targets will be the focus of future stages of research that determines how to best disrupt reproductive function without impacting growth performance.</p>
</section>
<section id="enhancing-sustainability-of-shellfish-aquaculture-through-streamlined-maturation-control" class="level3">
<h3 class="anchored" data-anchor-id="enhancing-sustainability-of-shellfish-aquaculture-through-streamlined-maturation-control">Enhancing sustainability of shellfish aquaculture through streamlined maturation control</h3>
<p><em>Summary</em>: The proposed objectives cover the essential first steps toward generating sterile bivalves via molecular disruption of germ cell formation. The first step in the process is the identification of genes involved in primordial germ cell (PGC) specification in bivalves via single-cell RNA sequencing (scRNA- Seq). This cutting-edge technique is particularly suited to identify germ cell markers in bivalves, since germ cell precursors represent a small number of cells in developing embryos. This information is integral to being able to control germ cell fate for reproductive sterilization of aquaculture species, and our proposed use of scRNA-Seq for directed application to aquaculture is both novel and potentially highly beneficial to the broader shellfish aquaculture community. Our first specific research aim is to characterize genomic processes involved in germ cell specification in Pacific oysters. A second fundamental step in the process is the optimization of techniques to deliver targeted, gene-regulating molecules to embryos to inhibit germ cell formation. Methods to optimize delivery techniques, including the use of CRISPR-Cas9 and morpholino constructs, have yet to be developed for bivalve embryos. Therefore, our second specific research aim is to optimize delivery techniques of custom gene-regulating molecules to oyster embryos. The completion of these objectives sets the stage for a more effective and sustainable approach for sterilization of shellfish.</p>
</section>
<section id="effects-of-temperature-change-and-hematodinium-sp.-infection-bitter-crab-disease-on-tanner-crab-chionoecetes-bairdi" class="level3">
<h3 class="anchored" data-anchor-id="effects-of-temperature-change-and-hematodinium-sp.-infection-bitter-crab-disease-on-tanner-crab-chionoecetes-bairdi">Effects of temperature change and Hematodinium sp. infection (Bitter Crab Disease) on Tanner crab (Chionoecetes bairdi)</h3>
<p><em>Summary</em>: Changing climate conditions, due to increasing releases of CO2 into the atmosphere, are causing warming of the world’s oceans. Changes in seawater temperatures are predicted to cause a shift in the distribution and a change in the abundance of most plants and animals. For most species, the magnitude of the impact, the potential for adaptation to future temperatures, and the mechanisms for adaptation are unknown. Features of parasite/disease ecology are also predicted to change as oceans warm, including susceptibility of hosts to disease, host ability to combat disease once infected, and alterations in pathogen virulence. Alaskan Tanner crab stocks, supporting fisheries worth $21 million in 2014, are expected to be significantly impacted directly and indirectly by warming temperatures. The Alaska Department of Fish and Game considers bitter crab disease, caused by a parasitic dinoflagellate, Hematodinium, to be the ‘principle threat’ to Alaskan Tanner crab stocks. Infection rates in the Bering Sea and southeast Alaska range from 2-5% and 0-100%, respectively. In heavily infected hosts, the meat is soft and chalky with a bitter taste, rendering the crabs unmarketable. The disease is believed fatal, although the time from infection to death remains uncertain. Recent worldwide spread of Hematodinium infections appears to have closely followed warming trends in the Atlantic and Pacific Oceans. We postulate that increased temperature in the North Pacific will physiologically stress Tanner crabs and also lead to increased prevalence of Hematodinium infections, either of which may lead to increased mortality in Tanner crabs. We propose to hold healthy and Hematodinium-infected Tanner crabs under different temperature regimes testing for a genetic response to infection and temperature. Our research will provide insight into the underlying mechanistic linkages between potential effects of climate change and important processes such as recruitment, growth and natural mortality on Alaskan Tanner crab stocks.</p>
<blockquote class="blockquote">
<h1 id="aquaculture-resource-development">Aquaculture Resource Development</h1>
</blockquote>
</section>
<section id="development-of-environmental-conditioning-practices-to-decrease-impacts-of-climate-change-on-shellfish-aquaculture" class="level3">
<h3 class="anchored" data-anchor-id="development-of-environmental-conditioning-practices-to-decrease-impacts-of-climate-change-on-shellfish-aquaculture">Development of environmental conditioning practices to decrease impacts of climate change on shellfish aquaculture</h3>
<p><em>Summary</em>: The project will be completed using commercial hatchery rearing of geoduck clams. We will test and implement shellfish performance enhancement methods through the use of environmental stressor preconditioning, or “hardening”. The objectives of this work are to A) Identify key stages in the geoduck life cycle when environmental conditioning can be applied for optimal benefits to productivity, including: - Broodstock conditioning and reproductive performance, - Larval growth and survivorship, and - Juvenile resistance to stress through repeated exposures, B) Use genomics and epigenomics to identify underlying mechanisms involved in enhanced performance. This will provide mechanistic information that enables the application of optimal preconditioning approaches in other shellfish species.<br>
This project directly addresses the research topics of genomics, quantitative genetics, and phenomics in the less studied, yet viable, geoduck clam. The proposed work is based on our group’s prior finding that some exposure to low pH conditions can improve larval and juvenile performance traits. The research proposed will integrate and further test these findings in a commercial shellfish hatchery setting and determine the genetic and epigenetic markers associated with improved performance. In addition, we will explore how the “memory” of prior exposure increases later performance such that epigenetic markers could be leveraged in broodstock management and hatchery practices. A number of phenotypes will be assessed, as well as the underlying genomic factors, including fecundity, survival, and growth.</p>
</section>
<section id="caliguslife-scientific-research-of-excellence-towards-understanding-sea-lice-biology-and-its-application-in-control-strategies-for-the-salmon-industry" class="level3">
<h3 class="anchored" data-anchor-id="caliguslife-scientific-research-of-excellence-towards-understanding-sea-lice-biology-and-its-application-in-control-strategies-for-the-salmon-industry">CaligusLIFE: Scientific research of excellence towards understanding sea lice biology and its application in control strategies for the salmon industry</h3>
<p><em>Summary</em>: This project will address specifically the generation of new knowledge in the areas of the Biological Cycle of Caligus and Genomics. The aim is to identify those environmental and genetic factors that significantly impact the life cycle of Caligus. This information will be used to generate mathematical models that calculate fecundity and production of Caligus and describe the spatio-temporal patterns of dispersion that explain the observed patterns of infestation in the field. The genomic part will include wide genome sequencing (WGS), assembly and functional annotation of Caligus and epigenomics.</p>
<section id="comparative-epigenomic-analyses-across-bivalve-genome-resources-ceabigr" class="level4">
<h4 class="anchored" data-anchor-id="comparative-epigenomic-analyses-across-bivalve-genome-resources-ceabigr">Comparative Epigenomic Analyses across Bivalve Genome Resources (CEABiGR)</h4>
<p><em>Summary</em>: Bivalve mollusks constitute the major players in US aquaculture production with oysters, mussels, and clams bringing in over $300 million in 2016 (NOAA Fisheries). There are several genomes of economically important bivalves with several other species’ genomes in the midst of completion. Further, numerous labs across the US are working on the functional annotations of these genomes including characterizing aspects of genetic, epigenetic and microbiome variation. There is a wealth of disparate information (WGS, RAD-Seq, bisulfite sequence data, metagenomics, and data sets associated with miRNAs, lncRNAs, and histone modifications) that when integrated would significantly advance our functional understanding of bivalve genomes. The integration of these datasets in a comparative fashion will improve the annotation of all bivalve genomes and provide valuable resources for the entire community. We are requesting funds to support a workshop for researchers to come together and focus on improving the assembly and annotation of bivalve genomes in a comparative framework. This workshop will be planned for 2020 and will be held at the University of Washington. The format of the workshop will be modeled after the very successful NRSP-8 supported workshop hosted by Dr.&nbsp;Hollie Putnam at the University of Rhode Island focused on the Eastern oyster (https:// github.com/hputnam/FROGER). The functional analysis methods developed during the FROGER meeting will be further developed for application across bivalve species.</p>
</section>
</section>
<section id="leveraging-transformative-omics-technologies-to-alleviate-barriers-to-us-shellfish-production-omics-tech" class="level3">
<h3 class="anchored" data-anchor-id="leveraging-transformative-omics-technologies-to-alleviate-barriers-to-us-shellfish-production-omics-tech">Leveraging transformative ’omics technologies to alleviate barriers to US shellfish production (Omics Tech)</h3>
<p><em>Summary</em>: Objectives: It is difficult to overstate the importance of triploid oysters to the commercial shellfish industry. Our goal is to uncover fundamental genomic and physiological differences between diploid and triploid Pacific oysters leading to low survival in subpar grow-out environments, and apply this information to increase sustainable commercial aquaculture production. The specific research objectives of our proposal are to: • Evaluate performance differentiation between triploid and diploid oysters under specific and combined stress conditions. • Identify and characterize physiological mechanisms and genomic infrastructure underlying poor performance in triploid oysters in hatchery and grow-out conditions.</p>
</section>
<section id="development-of-omics-and-bioinformatics-approaches-for-marine-organisms-in-support-of-research-in-aquaculture-ocean-acidification-and-fisheries-assessments-aqua-omics" class="level3">
<h3 class="anchored" data-anchor-id="development-of-omics-and-bioinformatics-approaches-for-marine-organisms-in-support-of-research-in-aquaculture-ocean-acidification-and-fisheries-assessments-aqua-omics">Development of ’omics and bioinformatics approaches for marine organisms in support of research in aquaculture, ocean acidification, and fisheries assessments (Aqua-Omics)</h3>
<p><em>Summary</em>: The rapid development of molecular genetic methodology over the last decade has led to transformative change in many areas of biology and medicine. In particular, the emergence of high-throughput, or next-generation, DNA sequencing technology has dramatically increased capacity for the generation of data a million-fold. Many of the traditional types of genetic analysis, including descriptions of population structure, trait mapping, individual identification and identification of stock boundaries and distributions, have been enabled by these new data sources in ways that were unimaginable several decades ago. These include detection of focal species in habitat with unknown occupancy, evaluation of species or stock composition in particular time/area strata of fishery management areas from water samples, the elucidation of trophic relationships between marine animals, their predators and prey, from gut contents or fecal matter, and understanding the genetic basis of environmentally-relevant, adaptive or aquaculture-relevant traits.<br>
In this project, we will add capacity to collaborative research between partners at the University of Washington and the Northwest Fisheries Science Center, addressing objectives of key importance to the National Marine Fisheries Service mission. These include contributions to the development of ’omics and bioinformatics workflows for both data generation and data analysis (bioinformatics). Though the focus of the project is varied in taxonomic interest, the fundamental basis of the project is tied together for similarities in how data are generated and subsequently processed for questions of interest. In this project, our goal is to contribute to bioinformatics tools and workflows for research relevant to species assessments, aquaculture, and ocean acidification research.</p>
</section>
<section id="development-of-innovative-approaches-to-support-sustainable-aquaculture-and-understand-the-effects-of-ocean-acidification-on-marine-species-vernon" class="level3">
<h3 class="anchored" data-anchor-id="development-of-innovative-approaches-to-support-sustainable-aquaculture-and-understand-the-effects-of-ocean-acidification-on-marine-species-vernon">Development of innovative approaches to support sustainable aquaculture and understand the effects of ocean acidification on marine species (Vernon)</h3>
<p><em>Summary</em>: Environmental variability due to climate change and ocean acidification (OA) are having multifaceted impacts on marine species, ecosystems and human industries such as marine aquaculture. It is absolutely critical to understand how marine organisms, and industries that rely on them, will respond to these changing conditions. Arguably, it will be just as important to provide tools and best practices to mitigate these effects, and in the case of marine aquaculture – find solutions to increase sustainability of the sector broadly, to create resilience even in the face of environmental uncertainty. Innovative scientific solutions are required to meet these ongoing and future challenges. In this project, we will add capacity to collaborative research between partners at the University of Washington and the Northwest Fisheries Science Center, addressing objectives of key importance to the National Marine Fisheries Service mission. These include innovative problem-solving approaches to address challenges facing sustainable aquaculture development, understanding the impacts of climate variability on marine fish and shellfish, and woven throughout these research activities is the development of ’omics and bioinformatics workflows for both data generation and data analysis (bioinformatics). Through this composite project, our goal is to contribute to applications to support sustainable aquaculture, increase our understanding of the effects of OA on marine fish and shellfish and contribute bioinformatic capacity, tools and workflows for research related to aquaculture, genomics and climate change.</p>
</section>
<section id="readying-sustainable-aquaculture-for-a-changing-ocean-uncovering-the-mechanisms-associated-with-intergenerational-carryover-effects-to-enhance-bivalve-resilience-to-acidification" class="level3">
<h3 class="anchored" data-anchor-id="readying-sustainable-aquaculture-for-a-changing-ocean-uncovering-the-mechanisms-associated-with-intergenerational-carryover-effects-to-enhance-bivalve-resilience-to-acidification">Readying sustainable aquaculture for a changing ocean: uncovering the mechanisms associated with intergenerational carryover effects to enhance bivalve resilience to acidification</h3>
<p><em>Summary</em>: The proposed effort will identify how parental environmental conditions drive shellfish offspring performance, including describing underlying mechanisms. To do this we will examine intergenerational effects of ocean acidification in the Manila clam, a globally cultured species that will serve as a model for marine bivalves and related taxa. By comprehensively investigating gamete status following parental pCO2 exposures, this project will uncover the non-genetic carriers of information across generations while simultaneously identifying factors that predict high performance of larvae in acidified conditions. By leveraging and working closely on a complementary NOAA supported project, we are able to expand our traditional research objectives with an objective centered around increasing diversity and inclusion in marine sciences. The specific objectives include 1) Characterize carryover performance in clams in response to ocean acidification, 2) Identify maternal macromolecule contribution of intergenerational plasticity and carryover performance, 3) Identify paternal epigenetic signatures associated with intergenerational plasticity and carryover performance, and 4) Develop inclusive educational experiences and products for underserved groups. The proposed work supports Washington Sea Grants commitment to cultivating partnerships and practicing a commitment to diversity, equity and inclusion. The objectives directly align with the critical program areas of Sustainable Fisheries and Aquaculture, Ocean Literacy and Workforce Development, and Healthy Coastal Ecosystems.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>